tests/integration/tests/test_etcd.py

```diff
@@ -0,0 +1,209 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import logging
+import textwrap
+from typing import List
+
+import pytest
+from test_util import harness, util
+
+LOG = logging.getLogger(__name__)
+
+ETCD_URL = "https://github.com/etcd-io/etcd/releases/download"
+ETCD_VERSION = "v3.3.8"
```

Comment:
any chance the ETCD_VERSION could be parameterized in  `tests/integration/tests/test_util/config.py`

+1 for an `os.getenv("ETCD_VERSION") or latest_etcd_release(...)`

so that we're always testing against the latest etcd

Reply:
you can set the `ETCD_VERSION` in the environment now.